src/test/scala/org/kibanaLoadTest/helpers/KbnClient.scala

```diff
@@ -0,0 +1,172 @@
+      if (loginResponse != null) loginResponse.close()
+      client.close()
+    }
+  }
+
+  def load(path: Path): Unit = {
+    val archivePath = this.getJsonPath(path)
+    checkFilesExist(archivePath)
+    val savedObjectStringList = Helper.readArchiveFile(archivePath)
+    logger.info(
+      s"Importing ${savedObjectStringList.length} saved objects from [${archivePath.toString}]"
+    )
+    val cookie = this.getCookie()
+    val importRequest = new HttpPost(
+      config.baseUrl + "/api/saved_objects/_import?overwrite=true"
+    )
+    importRequest.addHeader("Connection", "keep-alive")
+    importRequest.addHeader("kbn-version", config.buildVersion)
+    importRequest.addHeader("Cookie", cookie)
+
+    val builder = MultipartEntityBuilder.create
+    builder.addBinaryBody(
+      "file",
+      new ByteArrayInputStream(
+        savedObjectStringList.mkString("\n").getBytes(StandardCharsets.UTF_8)
+      ),
+      ContentType.APPLICATION_OCTET_STREAM,
+      "import.ndjson"
+    )
+    val multipart = builder.build()
+    importRequest.setEntity(multipart)
+    var response: CloseableHttpResponse = null
+    val client = HttpClientBuilder.create.build
+    try {
+      response = client.execute(importRequest)
+      val responseBody = EntityUtils.toString(response.getEntity)
+
+      val isSuccess = responseBody
+        .extract[Boolean](Symbol("success"))
+      val successCount = responseBody
+        .extract[Int](Symbol("successCount"))
+      if (isSuccess && (successCount == savedObjectStringList.length)) {
+        logger.info("Import finished successfully")
+      } else {
+        logger.error(
+          s"Import finished with errors, $successCount out of ${savedObjectStringList.length} saved objects imported"
+        )
+      }
+    } catch {
+      case e: Throwable =>
+        throw new RuntimeException("Exception during saved objects import", e)
+    } finally {
+      if (response != null) response.close()
+      client.close()
+    }
+  }
+
+  def unload(path: Path): Unit = {
+    val archivePath = this.getJsonPath(path)
+    checkFilesExist(archivePath)
+    val savedObjects = Helper
+      .readArchiveFile(archivePath)
+      .map(str => {
+        val json = parse(str).getOrElse(Json.Null)
+        val id = json.hcursor.get[String]("id").getOrElse(null)
+        val soType = json.hcursor.get[String]("type").getOrElse(null)
+        SavedObject(id, soType)
+      })
+
+    logger.info(
+      s"Deleting ${savedObjects.length} saved objects from [${archivePath.toString}]"
+    )
+    // login to Kibana once and reuse cookie for concurrent requests
+    val cookie = this.getCookie()
+    // Allow 20 requests to be executing at once
+    implicit val executionContext = ExecutionContext.fromExecutor(
+      new java.util.concurrent.ForkJoinPool(
+        20
+      )
+    )
+
+    val requestsFuture = Future.sequence(
+      savedObjects.map(so =>
+        Future {
+          val client = HttpClientBuilder.create.build
```

Comment:
Isn't it sufficient to create the client once for all requests? My understanding is that we create a new client for each request here.

Reply:
Initially I was trying to create it only once and re-use for concurrent requests, but it got stuck with only 2 requests out of 20 finished... and I think I need to use [PoolingHttpClientConnectionManager](https://hc.apache.org/httpcomponents-client-4.5.x/current/httpclient/apidocs/org/apache/http/impl/conn/PoolingHttpClientConnectionManager.html) but then I faced execution speed limit :( I will give it another shot.